### PR TITLE
working

### PR DESCRIPTION
**Pull request by bethanyhaubert**

### CODE REVIEW

**Comment by sumeetjain** [comment]
The solution here is slightly similar to the one for find-only-unique-elements. Let me know if you have questions about that, because it's possible you'll be able to make some more progress on this one after you work more on the other one.

**Comment by sumeetjain** [comment]
Your thought process seems pretty good to me. You might even try the code you've got.

**Comment by bethanyhaubert** [comment]
what do you mean 'try the code you've got'? i already tried out that code.

**Comment by sumeetjain** [comment]
Sorry, I realize that makes little sense. Your code almost works. There is a missing `end` somewhere that is causing an error when you run the code though. Find the missing `end`, and your tests should pass. Remember that proper indentation is usually the easiest way to discover missing `end`s.

File: lib/find_non_repeated_letter.rb [hunk 1]
```diff
@@ -8,7 +8,29 @@
...
+  #for each character in the array, if it does not have a previous occurence in the 'h' hash, add it to the hash with a value of '1'
+  arr.each do |n|
+    if h.has_key?(n) == false
+      h[n]=1
+  #if it does have a previous occurence in the 'h' hash, add '1' to the previously occuring value of that key
+    elsif h.has_key?(n) == true
+      h[n]= h[n] +1
+    end
+    #return the completed hash
+    return h
```
Comment (sumeetjain) [inline lib/find_non_repeated_letter.rb]: This `return h` is inside of the `arr.each` block. That means you're returning `h` one time for _each_ iteration of the loop.

You only need to do a `return` once–it's the final step of an algorithm.

So you've fixed the missing `end` problem, but the placement of that `end` is not correct.

**Comment by bethanyhaubert** [comment]
thanks

**Comment by sumeetjain** [comment]
:+1:

**Pull request closed without merging**